### ISSUE TITLE
Swift 2.2 support

### ISSUE THREAD

**Issue by NachoSoto** [ReactiveCocoa/ReactiveCocoa#2684]
I've started working on Swift 2.2 support (which we'll probably add in `RAC 4.1`). Right now I'm blocked by [this issue](https://github.com/Quick/Nimble/pull/243) in Nimble though.

**Comment by mdiep** [comment]
Will these changes be backwards compatible? If not, we should release these as `RAC 5.0`.

**Comment by NachoSoto** [comment]
Hopefully the code will! The built frameworks attached to the release won't... I don't really like the idea of bumping RAC's major version for every Swift minor version :/ I totally see what you're saying though.

Hopefully we can get it to compile with either, and make it clear that the attached frameworks are meant to be used with the latest version of Xcode.

**Comment by mdiep** [comment]
I'd be good with that. :+1: 

It's a pity that Swift doesn't use semantic versioning.

**Comment by joshbax** [comment]
FWIW, I was able to get ReactiveCocoa to build in Xcode 7.3 Beta 3. But I had to skip omit the `Nimble-tvOS` target. Is that the blocking issue?

**Comment by calebd** [comment]
Maybe I should open a separate issue for this. But I see a crash in ReactiveCocoa 4.0.1 / Swift 2.2 that I did not see before.

<img width="812" alt="screen shot 2016-02-27 at 12 35 29 pm" src="https://cloud.githubusercontent.com/assets/1761121/13375201/a5f21878-dd4e-11e5-8e4f-72cb3eb5c437.png">

**Comment by NachoSoto** [comment]
Ugh that sucks. Can you file a radar with a sample project? I can forward it to some Apple folks to look at.

**Comment by calebd** [comment]
@NachoSoto Sadly I was unable to reproduce this in an isolated app. Would you be able to tell me what pars of ReactiveCocoa I should focus on if I provide a call stack?

**Comment by calebd** [comment]
If it helps. I have tried several simple test cases and have had no luck. This issue still exists in beta 5.

<img width="802" alt="screen shot 2016-03-01 at 2 47 17 pm" src="https://cloud.githubusercontent.com/assets/1761121/13444704/a3251260-dfbc-11e5-94f4-f6ce7b87d4eb.png">

**Comment by calebd** [comment]
Disregard that last message. I got past the first crash and found another in a completely unrelated part of ReactiveCocoa. Seems like there is a bigger issue here.

**Comment by NachoSoto** [comment]
Release [`v4.1.0`](https://github.com/ReactiveCocoa/ReactiveCocoa/releases/tag/v4.1.0) compatible with `Xcode 7.3` and `Swift 2.x`.
